lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -114,3 +114,62 @@ function gutenberg_override_default_rest_server() {
 	return 'Gutenberg_REST_Server';
 }
 add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server', 1 );
+
+/**
+ * Updates the comment type in the REST API for WordPress version 6.7.
+ *
+ * This function is used as a filter callback for the 'rest_pre_insert_comment' hook.
+ * It checks if the 'comment_type' parameter is set to 'block_comment' in the REST API request,
+ * and if so, updates the 'comment_type' and 'comment_approved' properties of the prepared comment.
+ *
+ * @param array $prepared_comment The prepared comment data.
+ * @param WP_REST_Request $request The REST API request object.
+ * @return array The updated prepared comment data.
+ */
+if ( ! function_exists( 'update_comment_type_in_rest_api_6_7' ) && gutenberg_is_experiment_enabled( 'gutenberg-block-comment' ) ) {
+	function update_comment_type_in_rest_api_6_7( $prepared_comment, $request ) {
+		if ( ! empty( $request['comment_type'] ) && 'block_comment' === $request['comment_type'] ) {
+			$prepared_comment['comment_type']     = $request['comment_type'];
+			$prepared_comment['comment_approved'] = $request['comment_approved'];
+		}
+
+		return $prepared_comment;
+	}
+	add_filter( 'rest_pre_insert_comment', 'update_comment_type_in_rest_api_6_7', 10, 2 );
+}
+
+/**
+ * Updates the comment type for avatars in the WordPress REST API.
+ *
+ * This function adds the 'block_comment' type to the list of comment types
+ * for which avatars should be retrieved in the WordPress REST API.
+ *
+ * @param array $comment_type The array of comment types.
+ * @return array The updated array of comment types.
+ */
+if ( ! function_exists( 'update_get_avatar_comment_type' ) && gutenberg_is_experiment_enabled( 'gutenberg-block-comment' ) ) {
+	function update_get_avatar_comment_type( $comment_type ) {
+		$comment_type[] = 'block_comment';
+		return $comment_type;
+	}
+	add_filter( 'get_avatar_comment_types', 'update_get_avatar_comment_type', 10, 1 );
+}
+
+/**
+ * Updates the comment type filter dropdown options.
+ *
+ * This function is only defined if the 'gutenberg-block-comment' experiment is enabled and the 'update_comment_type_filter_dropdown' function does not already exist.
+ * It returns an array of comment type options for the comment type filter dropdown in the admin area.
+ *
+ * @return array An associative array of comment type options.
+ * The keys are the comment type slugs and the values are the translated names of the comment types.
+ */
+if ( ! function_exists( 'update_comment_type_filter_dropdown' ) && gutenberg_is_experiment_enabled( 'gutenberg-block-comment' ) ) {
+	function update_comment_type_filter_dropdown() {
+		return array(
+			'comment'       => __( 'Comments' ),
+			'block_comment' => __( 'Block Comments' ),
+		);
+	}
+	add_filter( 'admin_comment_types_dropdown', 'update_comment_type_filter_dropdown' );
```

Comment:
Why do we need to add this comment type to the comment management list? Comments are managed in the editor?

Reply:
We have incorporated the comment type into the backend comment page filter based on the feedback received: https://github.com/WordPress/gutenberg/pull/65181#discussion_r1751869616.